### ISSUE TITLE
Fix false positives for interpolation inside var in custom-property-pattern

### ISSUE THREAD

**Issue by yinonov** [stylelint/stylelint#5921]
### What steps are needed to reproduce the bug?

run stylelint on a sass file that interpolates sass variable like so
```css
background-color: var(#{appearance.$appearance-color-fill});
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "ignoreFiles": [
    "**/*"
  ],
  "overrides": [
    {
      "files": [
        "**/*.scss"
      ],
      "plugins": [
        "stylelint-no-unsupported-browser-features"
      ],
      "extends": [
        "stylelint-config-standard-scss",
        "stylelint-config-idiomatic-order",
        "stylelint-config-prettier"
      ],
      "rules": {
        "indentation": "tab",
        "plugin/no-unsupported-browser-features": [
          true,
          {
            "severity": "warning"
          }
        ],
        "order/properties-alphabetical-order": null,
        "selector-max-id": null,
        "selector-class-pattern": null,
        "max-nesting-depth": null,
        "selector-max-compound-selectors": null,
        "no-duplicate-selectors": true,
        "declaration-block-no-duplicate-properties": true,
        "function-parentheses-space-inside": null,
        "rule-empty-line-before": null
      }
    }
  ]
}
```

### How did you run Stylelint?

`npx nx run-many -- --target=stylelint --all`

### Which version of Stylelint are you using?

`14.5.1`

### What did you expect to happen?

stylelint `custom-property-pattern` rule should maybe ignore this pattern?

### What actually happened?

stylelint `custom-property-pattern` rule expects custom property name to be kebab-case
<img width="598" alt="image" src="https://user-images.githubusercontent.com/15140652/154837315-21554b88-9738-41d5-bbdb-11a9d4526eac.png">

### Does the bug relate to non-standard syntax?

Yes, it's related to scss interpolation

### Proposal to fix the bug

not familiar with the areas of responsibilities but might this be owned by `stylelint-config-standard-scss`?

**Comment by jeddy3** [comment]
@yinonov Thanks for the report and for using the template.

> not familiar with the areas of responsibilities but might this be owned by stylelint-config-standard-scss?

We'll handle it here in Stylelint as the built-in rules should ignore non-standard syntax like dollar variables inside of `var()` functions.

We can `isCustomProperty` and return early above:

https://github.com/stylelint/stylelint/blob/851c40b0a7fc4796036971906e21fff291c42649/lib/rules/custom-property-pattern/index.js#L48

e.g:

```js
const firstNodeValue = nodes[0].value;
if (!isCustomProperty(firstNodeValue)) return;
if (regexpPattern.test(firstNodeValue.slice(2))) return; 
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by yinonov** [comment]
Thanks for handing me out the opportunity. a PR has been linked to this issue. please let me know what you think
